# HG changeset patch
# User HackaLittleBit
# Date 1419935077 0
#      Tue Dec 30 10:24:37 2014 +0000
# Node ID a5d47307ce27e605b90e6b823880ad4480c69450
# Parent  9323d01f75f43940f7db51ec746576656265c681
Change: Seperate IsBuoy function in two.
IswaypointAir and Isbuoy.
Both with "facilities" check.
facilities FACIL_DOCK for ships.
facilities FACIL_AIRPORT for aircraft.

diff -r 9323d01f75f4 -r a5d47307ce27 src/station_map.h
--- a/src/station_map.h	Tue Dec 30 10:22:12 2014 +0000
+++ b/src/station_map.h	Tue Dec 30 10:24:37 2014 +0000
@@ -172,6 +172,10 @@
 
 bool IsHangar(TileIndex t);
 
+bool IsBuoy(TileIndex t);
+
+bool IsWaypointAir(TileIndex t);
+
 /**
  * Is the station at \a t a truck stop?
  * @param t Tile to check
@@ -299,17 +303,27 @@
 }
 
 /**
- * Is tile \a t a buoy tile?
+ * Is tile \a t a waypoint tile?
  * @param t Tile to check
  * @pre IsTileType(t, MP_STATION)
- * @return \c true if the tile is a buoy
+ * @return \c true if the tile is a waypoint.
  */
-static inline bool IsBuoy(TileIndex t)
+static inline bool IsWaypoint(TileIndex t)
 {
 	return GetStationType(t) == STATION_BUOY;
 }
 
 /**
+ * Is tile \a t an aircraft waypoint tile?
+ * @param t Tile to check
+ * @return \c true if the tile is a aircraft waypoint.
+ */
+static inline bool IsWaypointAirTile(TileIndex t)
+{
+	return IsTileType(t, MP_STATION) && IsWaypointAir(t);
+}
+
+/**
  * Is tile \a t a buoy tile?
  * @param t Tile to check
  * @return \c true if the tile is a buoy
diff -r 9323d01f75f4 -r a5d47307ce27 src/waypoint_cmd.cpp
--- a/src/waypoint_cmd.cpp	Tue Dec 30 10:22:12 2014 +0000
+++ b/src/waypoint_cmd.cpp	Tue Dec 30 10:24:37 2014 +0000
@@ -35,6 +35,28 @@
 #include "safeguards.h"
 
 /**
+ * Is tile \a t a buoy tile?
+ * @param t Tile to check
+ * @pre IsTileType(t, MP_STATION)
+ * @return \c true if the tile is a buoy.
+ */
+bool IsBuoy(TileIndex t)
+{
+	return IsWaypoint(t) && (Waypoint::GetByTile(t)->facilities & FACIL_DOCK);
+}
+
+/**
+ * Is tile \a t an air waypoint tile?
+ * @param t Tile to check
+ * @pre IsTileType(t, MP_STATION)
+ * @return \c true if the tile is a air waypoint.
+ */
+bool IsWaypointAir(TileIndex t)
+{
+	return IsWaypoint(t) && (Waypoint::GetByTile(t)->facilities & FACIL_AIRPORT);
+}
+
+/**
  * Update the virtual coords needed to draw the waypoint sign.
  */
 void Waypoint::UpdateVirtCoord()
